apps/cpp_rpc/rpc_server.cc

```diff
@@ -0,0 +1,363 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file rpc_server.cc
+ * \brief RPC Server implementation.
+ */
+
+#include <tvm/runtime/registry.h>
+
+#if defined(__linux__) || defined(__ANDROID__)
+#include <sys/select.h>
+#include <sys/wait.h>
+#endif
+#include <set>
+#include <iostream>
+#include <future>
+#include <thread>
+#include <chrono>
+#include <string>
+
+#include "rpc_server.h"
+#include "rpc_env.h"
+#include "rpc_tracker_client.h"
+#include "../../src/runtime/rpc/rpc_session.h"
+#include "../../src/runtime/rpc/rpc_socket_impl.h"
+#include "../../src/common/socket.h"
+
+#if defined(__linux__) || defined(__ANDROID__)
+static pid_t waitpid_eintr(int *status) {
+  pid_t pid = 0;
+  while ((pid = waitpid(-1, status, 0)) == -1) {
+    if (errno == EINTR) {
+      continue;
+    } else {
+      perror("waitpid");
+      abort();
+    }
+  }
+  return pid;
+}
+#endif
+
+namespace tvm {
+namespace runtime {
+
+/*!
+ * \brief RPCServer RPC Server class.
+ * \param host The hostname of the server, Default=0.0.0.0
```

Comment:
should these arguments be part of the constructor?

Reply:
These arguments will be used in other class functions, like Start / ListenLoopProc and so on.